connor/antifraud/blacklist_watcher.go

```diff
@@ -0,0 +1,49 @@
+package antifraud
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sonm-io/core/proto"
+)
+
+const (
+	minStep = time.Hour
+	maxStep = time.Hour * 24 * 7
+)
+
+type blacklistWatcher struct {
+	address     common.Address
+	till        time.Time
+	currentStep time.Duration
+	client      sonm.BlacklistClient
+}
+
+func (m *blacklistWatcher) Failure() {
+	m.till = time.Now().Add(m.currentStep)
+	m.currentStep *= 2
+	if m.currentStep > maxStep {
+		m.currentStep = maxStep
+	}
+}
+
+func (m *blacklistWatcher) Success() {
+	m.currentStep /= minStep
+	if m.currentStep < minStep {
+		m.currentStep = minStep
+	}
+}
+
+func (m *blacklistWatcher) Blacklisted() bool {
```

Comment:
Misleading name. It actually checks only current state, not the global one, i.e. such address can be in blacklist in the blockchain.